Makefile.common

```diff
@@ -87,14 +87,14 @@ ifeq ($(GOHOSTARCH),amd64)
         endif
 endif
 
-.PHONY: all
-all: precheck style staticcheck unused build test
-
 # This rule is used to forward a target like "build" to "common-build".  This
 # allows a new "build" target to be defined in a Makefile which includes this
 # one and override "common-build" without override warnings.
 %: common-% ;
 
+.PHONY: common-all
+common-all: precheck style check_license staticcheck unused test build
```

Comment:
At least for AlertManager, `build` needs to happen before `test`.

Reply:
That's odd, why is that? Maybe the alertmanager needs some dependency fixes?

Reply:
Because some tests spawn the `alertmanager` binary. Eventually it should be fixed but that's how it works today.

Reply:
updated. Thanks Simon for the info.

Reply:
Yea, that's valid.

Maybe we should make test depend on build explicitly by default.

Reply:
I would say keep it generic and if some repo (like the alertmanager) needs this that it can add this to the main Makefile.